optimum/onnxruntime/quantization.py

```diff
@@ -81,71 +84,91 @@ class ORTQuantizer(ABC):
-            feature (`str`):
-                Feature to use when exporting the model.
-            opset (`int`, *optional*):
-                ONNX opset version to export the model with.
-
+            use_auth_token (`str` or `bool`, *optional*):
+                Is needed to load models from a private repository
+            from_transformers (`bool`, *optional*, defaults to `False`):
+                Defines wether the provided `model_name_or_path` contains a vanilla Transformers checkpoint.
+                ORTQuantizer will then convert the model first
+            task (`str`, *optional*):
+                Transformers pipeline task for the model. Will be used to convert the model if needed.
+            file_name(`str`, *optional*):
+                Overwrites the default model file name from `"model.onnx"` to `file_name`. This allows you to load different model files from the same
+                repository or directory.
+            cache_dir (`Union[str, Path]`, *optional*):
+                Path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
         Returns:
             An instance of `ORTQuantizer`.
         """
-        preprocessor = get_preprocessor(model_name_or_path)
-        model_class = FeaturesManager.get_model_class_for_feature(feature)
-        model = model_class.from_pretrained(model_name_or_path)
-
-        return ORTQuantizer(preprocessor, model, feature, opset)
+        model_file_name = ONNX_WEIGHTS_NAME if file_name is None else file_name
+        # check if the user is loading a vanilla transformers and if a feature is available
+        if from_transformers and task is None:
+            raise ValueError("When using from_transformers, you need to provide a feature/task.")
+
+        # converts vanilla transformers to onnx if correct task is provided and model is not already in onnx
+        if from_transformers and task:
+            model_class = SUPPORTED_TASKS.get(task, None).get("class", None)[0]
+            if not model_class:
+                raise ValueError(f"Task {task} is not supported.")
+            onnx_model = model_class.from_pretrained(
+                model_name_or_path,
+                from_transformers=from_transformers,
+                use_auth_token=use_auth_token,
+                cache_dir=cache_dir,
+            )
+            return cls(onnx_model.model_save_dir.joinpath(onnx_model.latest_model_name))
+
+        # create ORTQuantizer based on the provided input
+        if isinstance(model_name_or_path, ORTModel):
+            return cls(model_name_or_path.model_save_dir.joinpath(model_file_name))
+        elif os.path.isdir(model_name_or_path):
+            if not isinstance(model_name_or_path, Path):
+                model_name_or_path = Path(model_name_or_path)
+            return cls(model_name_or_path.joinpath(model_file_name))
+        elif task is not None:
```

Comment:
I would explicitly add in the condition that `from_transformers` is `False` for clarity and give it as an argument as well (in order to not rely on default values).